lib/elixir/lib/code.ex

```diff
@@ -1095,7 +1099,16 @@ defmodule Code do
   defp do_fetch_docs(bin_or_path) do
     case :beam_lib.chunks(bin_or_path, [@docs_chunk]) do
       {:ok, {_module, [{@docs_chunk, bin}]}} ->
-        :erlang.binary_to_term(bin)
+        case :erlang.binary_to_term(bin) do
+          {:docs_v1, _, _, _, _, _, _} = chunk ->
+            chunk
+
+          tuple when is_tuple(tuple) and is_atom(elem(tuple, 0)) ->
+            {:error, {:unsupported_version, elem(tuple, 0)}}
+
+          unknown ->
```

Comment:
Since the `binary_to_term` could fail altogether, maybe we should do this:

```elixir
try do
  :erlang.binary_to_term(binary)
rescue
  _ -> {:error, {:invalid_chunk, binary}}
end
```

Maybe somebody could put invalid data into the chunk but, if they do that, they deserve the errors they will get (which is true for any other chunk). :)